service/storage_service.cc

```diff
@@ -3822,7 +3365,11 @@ future<> storage_service::join_cluster(sharded<db::system_distributed_keyspace>&
     for (auto& x : loaded_peer_features) {
         slogger.info("peer={}, supported_features={}", x.first, x.second);
     }
-    co_return co_await join_token_ring(sys_dist_ks, proxy, std::move(initial_contact_nodes), std::move(loaded_endpoints), std::move(loaded_peer_features), get_ring_delay());
+
+    auto task = co_await get_task_manager_module().make_and_start_task<node_ops::join_token_ring_task_impl>({}, "", *this,
+        sys_dist_ks, proxy, std::move(initial_contact_nodes), std::move(loaded_endpoints), std::move(loaded_peer_features),
+        get_ring_delay());
```

Comment:
On shutdown we need to make sure to flush any task related to storage-service.
/cc @xemul

Reply:
It happens in `storage_service::stop()` which calls `task_manager::module::stop`. A gate it closed so no new tasks are registered and abort is requested so the tasks that finished are immediately unregistered (without waiing task_ttl time)